signer/src/pubkey.rs

```diff
@@ -0,0 +1,152 @@
+//! Public key structures and algorithms
+//!
+//! Definition of public key structure and helpers
+
+use ark_ff::{BigInteger, PrimeField};
+use bs58;
+use core::fmt;
+use sha2::{Digest, Sha256};
+use std::ops::Neg;
+
+use crate::{BaseField, CurvePoint, FieldHelpers};
+
+/// Length of Mina addresses
+pub const MINA_ADDRESS_LEN: usize = 55;
+
+/// Public key
+#[derive(Copy, Clone, fmt::Debug, PartialEq, Eq)]
+pub struct PubKey(CurvePoint);
+
+impl PubKey {
+    /// Create a public key from curve point
+    pub fn new(point: CurvePoint) -> Self {
+        Self(point)
+    }
+
+    /// Deserialize Mina address into public key
+    pub fn from_address(address: &str) -> Result<Self, &'static str> {
+        if address.len() != MINA_ADDRESS_LEN {
+            return Err("Invalid address length");
+        }
+
+        let bytes = bs58::decode(address)
+            .into_vec()
+            .map_err(|_| "Invalid address encoding")?;
+
+        let (raw, checksum) = (&bytes[..bytes.len() - 4], &bytes[bytes.len() - 4..]);
```

Comment:
if bs58 is variable-length encoding, don't you want to check the length of `bytes` instead of `address`? As bytes might be different length eventhough address is MINA_ADDRESS_LEN?

Reply:
AFAIK MINA_ADDRESS_LEN is on the b58 encoding, but I can add an extra check for bytes.